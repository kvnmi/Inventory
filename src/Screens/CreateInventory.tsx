import React, { useState } from "react";
import { StyleSheet } from "react-native";

import Screen from "../components/Screen";
import {
  heightPixel,
  pixelSizeHorizontal,
  widthPixel,
} from "../config/normalise";
import { storeInventoryItem } from "../helpers/inventories";
import { IInventories } from "../interfaces/IInventory";
import Form from "../components/Form";
import Field from "../components/FormFields";
import SubmitButton from "../components/SubmitButton";
import Modal from "../components/Modal";
import { Image } from "react-native";
import { AppNavScreenProps } from "../Router/navTypes";
import AppText from "../components/AppText";
import colors from "../components/colors";
import { validationSchema } from "../config/formValidationSchema";
import useAuth from "../Auth/useAuth";

function ListingEditScreen({ navigation }: AppNavScreenProps<"CreateListing">) {
  const { user } = useAuth();

  const [failureModal, showFailureModal] = useState({
    active: false,
    message: "",
  });
  const [successModal, showSuccessModal] = useState<boolean>(false);
  const closeModal = () => showFailureModal({ ...failureModal, active: false });

  async function handleSubmit(data: IInventories) {
    const response = await storeInventoryItem(data, user!.uid);
    if (!response.status) {
      showFailureModal({
        active: true,
        message: response.message,
      });
    } else {
      showSuccessModal(true);
    }
  }

  return (
    <Screen style={styles.container}>
      <Image style={styles.image} source={require("../assets/logo-red.png")} />
      <AppText style={styles.text}>Add Inventory Item</AppText>
      <Form
        initialValues={{
          description: "",
          name: "",
          price: "",
          id: "",
          totalStock: "",
        }}
        onSubmit={handleSubmit}
        validationSchema={validationSchema}
      >
        <Field
          fieldName="name"
          icon="card-text"
          maxLength={225}
          placeholder="Name"
          autoCapitalize="sentences"
          autoCorrect={true}
        />

        <Field
          fieldName="price"
          icon="cash"
          maxLength={8}
          placeholder="Price"
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="numeric"
        />
        <Field
          icon="counter"
          maxLength={8}
          placeholder="Stock"
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="numeric"
          fieldName={"totalStock"}
        />
        <Field
          icon="card-text"
          placeholder="Description"
          autoCapitalize="sentences"
          numberOfLines={3}
          autoCorrect={true}
          fieldName={"description"}
        />

        <SubmitButton title="Submit" />
      </Form>
      <Modal
        isActive={failureModal.active}
        onClose={closeModal}
        onPress={closeModal}
        confirmText=""
        deleteText="Close"
        message={failureModal.message}
        imgType="failure"
      />
      <Modal
        isActive={successModal}
        onClose={() => showSuccessModal(false)}
        onPress={navigation.goBack}
        confirmText="View Listing"
        deleteText="Continue"
        message={"Inventory listing added successfully"}
      />
    </Screen>
  );
}

const styles = StyleSheet.create({
  container: {
    marginTop: 8,
    paddingHorizontal: pixelSizeHorizontal(15),
    alignItems: "center",
  },
  image: {
    height: heightPixel(80),
    width: widthPixel(80),
    marginVertical: pixelSizeHorizontal(35),
    alignSelf: "center",
  },
  text: {
    fontWeight: "bold",
    color: colors.blue,
  },
});

export default ListingEditScreen;
