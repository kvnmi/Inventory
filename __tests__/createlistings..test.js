import React from "react";
import renderer from "react-test-renderer";
import CreateInventory from "../src/Screens/CreateInventory";

describe("Create Listing Screen", () => {
  const route = {
    params: {
      item: 1,
    },
  };
  const navigation = { navigate: jest.fn() };

  it("matches snapshot", () => {
    const tree = renderer
      .create(<CreateInventory route={route} navigation={navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
