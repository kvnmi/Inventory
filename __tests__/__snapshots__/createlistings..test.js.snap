// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Listing Screen matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    [
      "top",
    ]
  }
  style={
    [
      {
        "flex": 1,
      },
      {
        "alignItems": "center",
        "marginTop": 8,
        "paddingHorizontal": 30,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      {
        "alignSelf": "center",
        "height": 132,
        "marginVertical": 70,
        "width": 160,
      }
    }
  />
  <Text
    style={
      [
        {
          "fontFamily": "Avenir",
          "fontSize": 32,
        },
        {
          "color": "#0e1f3f",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Add Inventory Item
  </Text>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#e4e8e5",
          "borderRadius": 10,
          "flexDirection": "row",
          "marginVertical": 10,
          "padding": 8,
          "paddingLeft": 13,
        },
        {
          "width": "100%",
        },
      ]
    }
  >
    <Text />
    <TextInput
      autoCapitalize="sentences"
      autoCorrect={true}
      maxLength={225}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Name"
      placeholderTextColor="#6e6969"
      style={
        {
          "color": "black",
          "fontFamily": "Avenir",
          "fontSize": 28,
          "width": "100%",
        }
      }
      value=""
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#e4e8e5",
          "borderRadius": 10,
          "flexDirection": "row",
          "marginVertical": 10,
          "padding": 8,
          "paddingLeft": 13,
        },
        {
          "width": "100%",
        },
      ]
    }
  >
    <Text />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="numeric"
      maxLength={8}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Price"
      placeholderTextColor="#6e6969"
      style={
        {
          "color": "black",
          "fontFamily": "Avenir",
          "fontSize": 28,
          "width": "100%",
        }
      }
      value=""
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#e4e8e5",
          "borderRadius": 10,
          "flexDirection": "row",
          "marginVertical": 10,
          "padding": 8,
          "paddingLeft": 13,
        },
        {
          "width": "100%",
        },
      ]
    }
  >
    <Text />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="numeric"
      maxLength={8}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Stock"
      placeholderTextColor="#6e6969"
      style={
        {
          "color": "black",
          "fontFamily": "Avenir",
          "fontSize": 28,
          "width": "100%",
        }
      }
      value=""
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#e4e8e5",
          "borderRadius": 10,
          "flexDirection": "row",
          "marginVertical": 10,
          "padding": 8,
          "paddingLeft": 13,
        },
        {
          "width": "100%",
        },
      ]
    }
  >
    <Text />
    <TextInput
      autoCapitalize="sentences"
      autoCorrect={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Description"
      placeholderTextColor="#6e6969"
      style={
        {
          "color": "black",
          "fontFamily": "Avenir",
          "fontSize": 28,
          "width": "100%",
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fc5c65",
        "borderRadius": 15,
        "height": 100,
        "justifyContent": "center",
        "marginVertical": 20,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 32,
        }
      }
    >
      Submit
    </Text>
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Modal
      animationType="none"
      deviceHeight={null}
      deviceWidth={null}
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      onSwipeComplete={[Function]}
      panResponderThreshold={4}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      supportedOrientations={
        [
          "portrait",
          "landscape",
        ]
      }
      swipeDirection={
        [
          "down",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    />
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Modal
      animationType="none"
      deviceHeight={null}
      deviceWidth={null}
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      onSwipeComplete={[Function]}
      panResponderThreshold={4}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      statusBarTranslucent={false}
      supportedOrientations={
        [
          "portrait",
          "landscape",
        ]
      }
      swipeDirection={
        [
          "down",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    />
  </View>
</RNCSafeAreaView>
`;
